apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -231,7 +241,7 @@ export const onSaveComplete =
       )
         ? currentAiCustomizations[typedProperty]
         : 'NULL';
-      if (currentSaveType) {
+      if (currentSaveType && userHasAichatAccess) {
```

Comment:
Wonder if it might be handy to create a little thunk action that checks `userHasAichatAccess` and then calls the analytics reporter? This would be basically the same as the function you already created in `AichatView`. That way, callers can call this action directly and we don't need to remember to check `userHasAichatAccess` in all these places. 

```
export const sendAnalytics = (event: string, properties: object) => (dispatch: AppDispatch, getState: () => RootState) {
  const userHasAichatAccess = getState().aichat.userHasAichatAccess;
  if (userHasAichatAccess) {
    analyticsReporter.sendEvent(event, properties, PLATFORMS.BOTH);
  }
}
```

Edit: since certain events (submitting chat messages) don't require user access, we can also add an optional param like `skipAccessCheck = false` (or different name) to override the `userHasAichatAccess` check.

Reply:
Ooo..I like this idea. Updated and thanks!